hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -199,7 +199,10 @@ public void commitMigration(PartitionMigrationEvent event) {
             // has been already started this call will do nothing
             recordStore.startLoading();
         }
-        mapServiceContext.nullifyOwnedPartitions();
+
+        if (event.getCurrentReplicaIndex() == 0 || event.getNewReplicaIndex() == 0) {
```

Comment:
Why this check is added? Why we don't refresh if it's backup replica?

Reply:
Replica index of ownership is zero.
To refresh only owned partitions on a node i put that check. This is the reason.